docker_images/peft/app/pipelines/text_generation.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import os
+
+import torch
+from app import idle, timing
+from app.pipelines import Pipeline
+from huggingface_hub import model_info
+from peft import PeftModel
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+
+logger = logging.getLogger(__name__)
+
+
+class TextGenerationPipeline(Pipeline):
+    def __init__(self, model_id: str):
+        use_auth_token = os.getenv("HF_API_TOKEN")
+        model_data = model_info(model_id, token=use_auth_token)
+        config_dict = model_data.config["peft"]
```

Comment:
Maybe safer to use `.get`, which would return `None` rather than erroring out

```suggestion
        config_dict = model_data.config.get("peft")
```